packages/shared/lib/services/connection.service.ts

```diff
@@ -480,15 +480,28 @@ class ConnectionService {
         return encryptionManager.decryptConnection(res[0]!);
     }
 
-    public async setRefreshFailure({ id, currentAttempt }: { id: number; currentAttempt: number }) {
+    public async setRefreshFailure({ id, lastRefreshFailure, currentAttempt }: { id: number; lastRefreshFailure?: Date | null; currentAttempt: number }) {
+        let attempt = currentAttempt;
+        const now = new Date();
+
+        // Only increment once per day to avoid burst failed refresh invalidating a connection (e.g: provider being down)
+        if (
+            lastRefreshFailure &&
+            (lastRefreshFailure.getFullYear() < now.getFullYear() ||
+                lastRefreshFailure.getMonth() < now.getMonth() ||
+                lastRefreshFailure.getDate() < now.getDate())
+        ) {
+            attempt += 1;
+        }
```

Comment:
Do i understand correctly that attempt will stay at 0 for the entire first day it is failing?
Does it also mean if it fail first at 23:59 and then again at 00:01, attempt will be 1?

Reply:
1/ yes, but maybe it should be 1 you are right
2/ yes that is a limitation since we don't store all refresh date, but since it will stay N for the next 24 hours it's okay-ish imo